### PR TITLE
fix - in multhead_attention, correct the location where key & value shapes are checked

### PR DESCRIPTION
**Pull request by ksachdeva**
at present this check is done only if mask is not None whereas this should be checked irrespective of the msk

### CODE REVIEW

File: tensorflow_addons/layers/multihead_attention.py [hunk 1]
```diff
@@ -189,6 +189,11 @@ def call(self, inputs, training=None, mask=None):
         key = inputs[1]
         value = inputs[2] if len(inputs) > 2 else key
 
+        if key.shape[-2] != value.shape[-2]:
+            raise ValueError(
+                "the number of elements in 'key' must be equal to the same as the number of elements in 'value'"
+            )
+
         # verify shapes
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:197]: we might also want to move this comment upwards

Comment (ksachdeva) [reply tensorflow_addons/layers/multihead_attention.py]: done

**Comment by ksachdeva** [comment]
Hi,
I did run the checks as recommended and even had the hoot installed as per the instructions. Not sure what is wrong here.

**Comment by WindQAQ** [comment]
> Hi,
> I did run the checks as recommended and even had the hoot installed as per the instructions. Not sure what is wrong here.

No worries. GitHub CI sometimes fails due to network issue. Let me try to re-trigger it again.

**Pull request merged**